src/oldtimers/models.py

```diff
@@ -1,3 +1,296 @@
+    )
+    vin = models.CharField(null=False, blank=False, unique=True, max_length=17, primary_key=True, db_index=True)
+    category = models.PositiveSmallIntegerField(
+        choices=VEHICLE_CATEGORY_CHOICES.choices,
+        default=VEHICLE_CATEGORY_CHOICES.OTHER,
+        null=False,
+        blank=True,
+    )
+    brand = models.CharField(
+        max_length=50,
+        null=False,
+        blank=True,
+    )
+    model = models.CharField(
+        max_length=50,
+        null=False,
+        blank=True,
+    )
+    production_year = models.PositiveSmallIntegerField(
+        _("year"),
+        null=True,
+        blank=True,
+        validators=[MaxValueValidator(YEAR_OF_PRODUCTION_MAX_YEAR), MinValueValidator(YEAR_OF_PRODUCTION_MIN_YEAR)],
+    )
+    condition = models.PositiveSmallIntegerField(
+        choices=VEHICLE_CONDITION_CHOICES.choices,
+        default=VEHICLE_CONDITION_CHOICES.OTHER,
+        null=True,
+        blank=True,
+    )
+    image = models.ImageField(
+        default="default.png",
+        upload_to="media/vehicle",
+        null=True,
+        blank=True,
+    )
+    price = models.DecimalField(
+        max_digits=19, decimal_places=2, null=True, blank=True, validators=[MinValueValidator(1.00)]
+    )
+    description = models.TextField(
+        max_length=500,
+        null=False,
+        blank=True,
+    )
+
+    def __str__(self):
+        return (
+            f"{self.retailer.company_name} "
+            f"{self.brand} {self.model}"
+            f" {self.vin} {self.description} "
+            f"Price: {self.price} $ "
+        )
+
+    def save(self, *args, **kwargs):
+        self.full_clean()
+        return super().save(*args, **kwargs)
+
+
+class Retailer(BaseModel):
+    uuid = models.UUIDField(default=uuid.uuid4, db_index=True, unique=True)
+    company_name = models.CharField(max_length=50, blank=True, null=False)
+    mobile = PhoneNumberField(blank=True, null=False)
+    email = models.EmailField(max_length=128)
+    address = models.CharField(_("address"), max_length=128, null=False, blank=True)
+    city = models.CharField(_("city"), max_length=64, blank=True, null=False)
+    country = CountryField(blank=True, null=False)
+    zip_code = models.CharField(_("zip code"), max_length=5, blank=True, null=False)
+    service_fee = models.DecimalField(
+        max_digits=19,
+        decimal_places=2,
+        null=False,
+        blank=True,
+        default=1.00,
+        validators=[MinValueValidator(1.00)],
+    )
+
+    def __str__(self):
+        return f"{self.company_name} Id: {self.id}  Service fee: {self.service_fee} "
+
+    def vehicles_count(self):
+        return self.vehicles.count()
+
+    def save(self, *args, **kwargs):
+        self.full_clean()
+        return super().save(*args, **kwargs)
+
+
+class Offer(BaseModel):
+    retailer = models.ForeignKey(to="oldtimers.Retailer", related_name="offers", on_delete=models.CASCADE)
+    vehicle = models.OneToOneField(
```

Comment:
looks like useless model.  Offer_price you can calculate on a "fly". In vehicle model, you have FK to the retailer.